synapse/handlers/register.py

```diff
@@ -143,10 +143,15 @@ async def check_username(
         assigned_user_id: Optional[str] = None,
         inhibit_user_in_use_error: bool = False,
     ) -> None:
```

Comment:
As an aside -- I was surprised to see the appservice registration code path seems to completely skip `check_username`? Is that a bug?

Reply:
This seems to be #10246.